### PR TITLE
 [Feat] Product - Product Id modification

### PR DESCRIPTION
**Pull request by Juninhotech**
# Description
Base on the [comment ](https://github.com/hngprojects/hng_boilerplate_csharp_web/pull/92#discussion_r1688642345)on [PR](https://github.com/hngprojects/hng_boilerplate_csharp_web/pull/92) to change the product id from guid to database generated id, the owner asked me to pick it up due to some technical issues with his system.

# Changes proposed
Modifying product id 

## What were you told to do?
To change product id from guid to database generated (int)

## What did you do?
I modified the product id from (guid) to int

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
image
image

### CODE REVIEW

File: src/Hng.Domain/Entities/Product.cs [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+    {
+        CreatedAt = DateTime.UtcNow;
+    }
+
+    [Key]
+    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+    public new int Id { get; set; }
     public string Name { get; set; }
     public string Description { get; set; }
+    public string[] Category { get; set; }
```
Comment (Aniyikaye-Alabi) [inline src/Hng.Domain/Entities/Product.cs:18]: Having a prop with as an array can be tricky in my opinion. Create another entity for a category, properly specifying the relationship will be better.

**Review by Aniyikaye-Alabi — CHANGES_REQUESTED** [review]

File: src/Hng.Domain/Entities/Product.cs (hunk 1 above)
Comment (Juninhotech) [reply src/Hng.Domain/Entities/Product.cs:18]: Done

**Comment by buka4rill** [comment]
CleanShot 2024-07-24 at 10 36 26
this is wrong branch naming

**Comment by buka4rill** [comment]
Wrong PR naming convention

**Pull request closed without merging**